django/db/models/operators.py

```diff
@@ -0,0 +1,55 @@
+from django.core.exceptions import FieldError
+from django.db.models.expressions import CombinedExpression, DurationValue
+from django.db.models.fields import (
+    CharField, DateField, DateTimeField, DurationField, TextField, TimeField,
+)
+from django.db.models.functions import ConcatPair
+
+# Don't export anything, the only purpose of this module is to register
+# dispatch expressions
+__all__ = ()
+
+
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateField, DateField, commutative=False)
+@CombinedExpression.dispatch(CombinedExpression.SUB, DateTimeField, DateTimeField, commutative=False)
+@CombinedExpression.dispatch(CombinedExpression.SUB, TimeField, TimeField, commutative=False)
+class TemporalSubtraction(CombinedExpression):
+    def __init__(self, lhs, connector, rhs, output_field=None):
+        super(TemporalSubtraction, self).__init__(lhs, self.SUB, rhs,
```

Comment:
There are other examples of this indentation style throughout, I won't comment on each one. Also, I realise my preference for style is kind of arbitrary, but it *is* mostly consistent through the code base already.